### PR TITLE
[refactor] refactor hide exocirsm handler

### LINKED ISSUE #213: [Refactor] Hiding and exorcism player states

### PR DESCRIPTION
**Pull request by naufalhilmi12345**
### Main changes

- Implement state pattern for hide and exorcism handler

### Tasks List
[close #213]

### Attachments
All test pass:
image

### CODE REVIEW

**Review by rafiftaris — CHANGES_REQUESTED** [review]
move GM's enum to HudManager and InputManager

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 1]
```diff
@@ -25,69 +24,81 @@ private enum _playerState
...
+
+    public void InvokeExorcismState()
+    {
+        SendHudEvent(_playerState.Exorcism.ToString(), true);
+        SendPlayerStateEvent(_playerState.Exorcism.ToString());
+    }
+
+    public void ResetPlayerState()
     {
         //Debug.Log("[INVOKE PLAYER STATE] Player state: " + state);
```
Comment (RaiRiz) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:93]: delete this line since it won't be used again

Comment (RaiRiz) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:93]: delete this line since it won't be used again (no "state" var argument anymore)

Comment (naufalhilmi12345) [reply Assets/Astralization/Scripts/Managers/GameManager.cs:93]: how about if i change the debug log to "[INVOKE PLAYER STATE] Player state" insteadof delete that line?

Comment (RaiRiz) [reply Assets/Astralization/Scripts/Managers/GameManager.cs:93]: that would be good too

File: Assets/Astralization/Scripts/Player/InputManager.cs [hunk 2]
```diff
@@ -12,6 +12,13 @@ public class InputManager : StateMachine
     [SerializeField]
     private PlayerInput _playerInput;
     private PlayerState tempCurrentState;
```
Comment (RaiRiz) [inline Assets/Astralization/Scripts/Player/InputManager.cs:14]: should be `private PlayerState _tempCurrentState` instead; don't forget to check anything that references to this var

**Review by RaiRiz — APPROVED** [review]
all good from me!

**Review by uciii — APPROVED** [review]
looks good

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
I think it is better to put the states enum in Utils. What do you guys think?

**Review by rafiftaris — APPROVED** [review]
mantaappp

**Review by raihansyahdean — APPROVED** [review]
Nice

**Pull request merged**